### PR TITLE
add opa gatekeeper

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related: https://github.com/wolfi-dev/os/pull/1983

### Pre-review Checklist

- [ ] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [ ] Include tests, sufficient enough that you would trust this image running in production.
- [ ] The version included is the latest GA version of the software
- [ ] The latest tag points to the newest stable version
- [ ] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [x] The image runs as `nonroot` and GID/UID are set to 65532
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [x] ENTRYPOINT
  - [ ] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [ ] For base images leave empty
  - [ ] For dev variants set to entrypoint script that falls back to system
- [ ] CMD:
  - [ ] For server applications give arguments to start in daemon mode (may be empty)
  - [ ] For utilities/tooling bring up help e.g. `–help`
  - [ ] For base images with a shell, call it e.g. [/bin/sh]
- [ ] Consider where and how the image deviates from popular alternatives. Is there a good reason and is it documented?
- [x] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [ ] Check if environment variables are needed e.g. to set data locations
- [ ] Ensure the image responds to SIGTERM
  - [ ] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [ ] Documentation. Let's make this excellent. Include usage example.
- [ ] Error logs write to stderr and normal logs to stdout. DO NOT write to file.

### CODE REVIEW

**Comment by amouat** [comment]
Nice test!

File: images/gatekeeper/configs/latest.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - groupname: nonroot
+      gid: 65532
+  users:
+    - username: nonroot
+      uid: 65532
+      gid: 65532
+  run-as: 65532
+
+entrypoint:
+  command: manager
```
Comment (amouat) [inline images/gatekeeper/configs/latest.apko.yaml:22]: If this works, it's fine, but I'm surprised we don't need the full path?

Could also add a cmd with `--help` if it has one?

File: images/gatekeeper/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+| **Status** | stable |
+| **OCI Reference** | `cgr.dev/chainguard/gatekeeper` |
+| **Variants/Tags** | ![](https://storage.googleapis.com/chainguard-images-build-outputs/summary/gatekeeper.svg) |
+
+*[Contact Chainguard](https://www.chainguard.dev/chainguard-images) for enterprise support, SLAs, and access to older tags.*
+
+---
+<!--monopod:end-->
+
+Minimalist Wolfi-based Gatekeeper which is a policy controller for Kubernetes
```
Comment (amouat) [inline images/gatekeeper/README.md:14]: We need some usage docs, even if it just points to how to use it in a Helm a chart, or however it's normally run.

**Review by amouat — CHANGES_REQUESTED** [review]
Looks good, but missing docs and it would be worth double-checking entrypoint & cmd.

File: images/gatekeeper/README.md (hunk 2 above)
Comment (developer-guy) [reply images/gatekeeper/README.md:14]: Thanks @amouat, I've added some usage details and updated the cmd with help command.

**Review by amouat — COMMENTED** [review]
Could we add a short example of how to override the image in the upstream Helm chart?

Also could we do a one-liner using kustomize for the yaml install?

**Review by amouat — APPROVED** [review]

**Pull request merged**